.github/workflows/yarn-upgrade.yml

```diff
@@ -46,13 +46,13 @@ jobs:
         # We special-case typescript because it's not semantically versionned
         run: |-
           # Upgrade dependencies at repository root
```

Comment:
Is it safe to say that this will not be upgrading major versions for regular dependencies? If not, should we be?

For example, I believe we want to avoid upgrading the major version for `constructs`, yes?

Reply:
We can't upgrade major versions without due diligence. I think it's risky and can cause breaking changes to our users.

Reply:
Merged https://github.com/aws/aws-cdk/pull/10409 and udated to only auto-increment minor versions where appropriate.